django/db/models/indexes.py

```diff
@@ -185,58 +189,77 @@ def __eq__(self, other):
 
 
 class IndexExpression(Func):
-    """
-    Applies correct order and wrapping of expressions suitable for CREATE INDEX statements
-    """
+    """Order and wrap expressions for CREATE INDEX statements."""
     template = '%(expressions)s'
     wrapper_classes = (OrderBy, Collate)
 
+    def __init__(self, *expressions, output_field=None, connection=None, **extra):
+        self.connection = connection
+        super().__init__(*expressions, output_field=output_field, **extra)
```

Comment:
We cannot pass `connection` in the `__init__()` because it breaks e.g. `deepcopy()`. I decided to add a separate hook `set_wrapper_classes()`, it is not the cleanest solution, but it's not the end of the world :globe_with_meridians: , and we don't have much time before the feature freeze :clock1: .

Reply:
Mutating the class variable `wrapper_classes` feels a bit shaky if multiple backends are used in the same application. We could call the class variable something like `default_wrapper_classes` and use it to set an instance variable which we can mutate on `set_wrapper_classes`.

Reply:
But we don't mutate `IndexExpression.wrapper_classes`.

Reply:
Sorry, got confused with the `register_wrappers` classmethod. NVM!